From 9bd9e1d87c6611349bc3e85b21592db8be161e33 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Wed, 12 Apr 2023 11:24:55 +0200
Subject: [PATCH] Fix build with gcc-13

:Release Notes:

:Detailed Notes:
Fixes:

  In file included from ../../git/src/third_party/openscreen/src/discovery/dnssd/public/dns_sd_txt_record.cc:5:
  ../../git/src/third_party/openscreen/src/discovery/dnssd/public/dns_sd_txt_record.h:21:61: error: 'uint8_t' was not declared in this scope
     21 |   using ValueRef = std::reference_wrapper<const std::vector<uint8_t>>;
        |                                                             ^~~~~~~
  ../../git/src/third_party/openscreen/src/discovery/dnssd/public/dns_sd_txt_record.h:15:1: note: 'uint8_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
     14 | #include "platform/base/error.h"
    +++ |+#include <cstdint>
     15 |

  In file included from ../../git/src/third_party/openscreen/src/util/base64.cc:5:
  ../../git/src/third_party/openscreen/src/util/base64.h:19:37: error: ISO C++ forbids declaration of 'type name' with no type [-fpermissive]
     19 | std::string Encode(absl::Span<const uint8_t> input);
        |                                     ^~~~~~~

and many more:

  src/base/check.h:62:34: error: expected primary-expression before '(' token
  src/base/debug/profiler.cc:43:1: error: 'ReturnAddressLocationResolver' does not name a type
  src/base/debug/profiler.h:56:21: error: 'ReturnAddressLocationResolver' was not declared in this scope
  src/base/debug/profiler.h:56:9: error: ISO C++ forbids declaration of 'uintptr_t' with no type [-fpermissive]
  src/base/debug/profiler.h:56:9: error: typedef 'base::debug::uintptr_t' is initialized (use 'decltype' instead)
  src/base/debug/profiler.h:69:13: error: 'ReturnAddressLocationResolver' does not name a type
  src/cc/trees/target_property.cc:19:58: error: 'uint32_t' was not declared in this scope
  src/components/autofill/core/browser/autofill_ablation_study.cc:152:15: error: no declaration matches 'autofill::AblationGroup autofill::AutofillAblationStudy::GetAblationGroupImpl(const GURL&, const base::Time&, uint32_t) const'
  src/components/autofill/core/browser/autofill_ablation_study.h:64:38: error: 'uint32_t' has not been declared
  src/components/feature_engagement/internal/event_storage_validator.h:29:27: error: 'uint32_t' has not been declared
  src/components/feature_engagement/internal/event_storage_validator.h:30:27: error: 'uint32_t' has not been declared
  src/components/feature_engagement/internal/never_event_storage_validator.cc:19:45: error: 'uint32_t' has not been declared
  src/components/feature_engagement/internal/never_event_storage_validator.cc:20:45: error: 'uint32_t' has not been declared
  src/components/feature_engagement/internal/never_event_storage_validator.h:29:19: error: 'uint32_t' has not been declared
  src/components/feature_engagement/internal/never_event_storage_validator.h:30:19: error: 'uint32_t' has not been declared
  src/components/metrics/psi_memory_parser.h:109:3: error: 'uint32_t' does not name a type
  src/components/metrics/psi_memory_parser.h:56:36: error: expected ')' before 'period'
  src/components/metrics/psi_memory_parser.h:70:40: error: 'uint8_t' does not name a type
  src/components/metrics/psi_memory_parser.h:71:34: error: 'uint32_t' has not been declared
  src/components/metrics/psi_memory_parser.h:75:3: error: 'uint32_t' does not name a type
  src/components/metrics/psi_memory_parser_linux.cc:142:19: error: no declaration matches 'metrics::ParsePSIMemStatus metrics::PSIMemoryParser::ParseMetrics(const uint8_t*, uint32_t, int*, int*)'
  src/components/metrics/psi_memory_parser_linux.cc:44:1: error: no declaration matches 'metrics::PSIMemoryParser::PSIMemoryParser(uint32_t)'
  src/components/metrics/psi_memory_parser_linux.cc:58:10: error: no declaration matches 'uint32_t metrics::PSIMemoryParser::GetPeriod() const'
  src/components/password_manager/core/browser/generation/password_generator.h:20:14: error: 'uint32_t' does not name a type
  src/components/viz/common/shared_element_resource_id.cc:20:76: error: no matching function for call to 'viz::SharedElementResourceId::SharedElementResourceId(int)'
  src/components/viz/common/shared_element_resource_id.cc:24:41: error: no matching function for call to 'viz::SharedElementResourceId::SharedElementResourceId(const uint32_t&)'
  src/components/viz/common/shared_element_resource_id.cc:28:1: error: no declaration matches 'viz::SharedElementResourceId::SharedElementResourceId(uint32_t)'
  src/components/viz/common/shared_element_resource_id.cc:31:10: error: 'id_' was not declared in this scope; did you mean 'id_t'?
  src/components/viz/common/shared_element_resource_id.cc:35:61: error: 'id_' was not declared in this scope; did you mean 'id_t'?
  src/components/viz/common/shared_element_resource_id.h:22:44: error: expected ')' before 'id'
  src/components/viz/common/shared_element_resource_id.h:29:12: error: 'id_' was not declared in this scope; did you mean 'id_t'?
  src/components/viz/common/shared_element_resource_id.h:29:21: error: 'const class viz::SharedElementResourceId' has no member named 'id_'
  src/components/viz/common/shared_element_resource_id.h:34:67: error: 'id_' was not declared in this scope; did you mean 'id_t'?
  src/components/viz/common/shared_element_resource_id.h:34:75: error: 'const class viz::SharedElementResourceId' has no member named 'id_'
  src/components/viz/common/shared_element_resource_id.h:39:3: error: 'uint32_t' does not name a type
  src/components/viz/common/shared_element_resource_id.h:42:3: error: 'uint32_t' does not name a type
  src/device/base/synchronization/one_writer_seqlock.h:71:35: error: 'uintptr_t' in namespace 'std' does not name a type
  src/device/base/synchronization/one_writer_seqlock.h:71:35: error: 'uintptr_t' is not a member of 'std'
  src/device/base/synchronization/one_writer_seqlock.h:72:35: error: 'uintptr_t' in namespace 'std' does not name a type
  src/device/base/synchronization/one_writer_seqlock.h:72:35: error: 'uintptr_t' is not a member of 'std'
  src/device/base/synchronization/one_writer_seqlock.h:87:35: error: 'uintptr_t' in namespace 'std' does not name a type
  src/device/base/synchronization/one_writer_seqlock.h:87:35: error: 'uintptr_t' is not a member of 'std'
  src/device/base/synchronization/one_writer_seqlock.h:88:35: error: 'uintptr_t' in namespace 'std' does not name a type
  src/device/base/synchronization/one_writer_seqlock.h:88:35: error: 'uintptr_t' is not a member of 'std'
  src/device/bluetooth/public/cpp/bluetooth_uuid.cc:115:22: error: no declaration matches 'std::vector<unsigned char> device::BluetoothUUID::GetBytes() const'
  src/device/bluetooth/public/cpp/bluetooth_uuid.h:89:15: error: 'uint8_t' was not declared in this scope
  src/device/bluetooth/public/cpp/bluetooth_uuid.h:89:22: error: template argument 1 is invalid
  src/device/bluetooth/public/cpp/bluetooth_uuid.h:89:22: error: template argument 2 is invalid
  src/device/bluetooth/public/cpp/bluetooth_uuid.h:89:8: error: '<expression error>' in namespace 'std' does not name a type
  src/extensions/common/constants.cc:109:15: error: 'kWebstoreSignaturesPublicKey' was not declared in this scope; did you mean 'kWebstoreSignaturesPublicKeySize'?
  src/extensions/common/constants.cc:81:7: error: 'uint8_t' does not name a type
  src/extensions/common/constants.h:145:32: error: 'uint8_t' does not name a type
  src/extensions/renderer/bindings/api_invocation_errors.cc:107:64: error: 'std::string extensions::api_errors::IndexError(uint32_t, const std::string&)' redeclared as different kind of entity
  src/extensions/renderer/bindings/api_invocation_errors.h:54:24: error: 'uint32_t' was not declared in this scope
  src/extensions/renderer/bindings/api_invocation_errors.h:54:40: error: expected primary-expression before 'const'
  src/gpu/config/gpu_feature_info.h:62:15: error: 'uint32_t' was not declared in this scope
  src/gpu/config/gpu_feature_info.h:62:23: error: template argument 1 is invalid
  src/gpu/config/gpu_feature_info.h:62:23: error: template argument 2 is invalid
  src/gpu/config/gpu_feature_info.h:62:8: error: '<expression error>' in namespace 'std' does not name a type
  src/gpu/config/gpu_feature_info.h:64:15: error: 'uint32_t' was not declared in this scope
  src/gpu/config/gpu_feature_info.h:64:23: error: template argument 1 is invalid
  src/gpu/config/gpu_feature_info.h:64:23: error: template argument 2 is invalid
  src/gpu/config/gpu_feature_info.h:64:8: error: '<expression error>' in namespace 'std' does not name a type
  src/net/base/parse_number.h:102:5: error: 'uint32_t' has not been declared
  src/net/base/parse_number.h:107:5: error: 'uint64_t' has not been declared
  src/net/cookies/cookie_inclusion_status.cc:292:6: error: no declaration matches 'bool net::CookieInclusionStatus::ValidateExclusionAndWarningFromWire(uint32_t, uint32_t)'
  src/net/cookies/cookie_inclusion_status.h:348:51: error: 'uint32_t' has not been declared
  src/net/cookies/cookie_inclusion_status.h:349:51: error: 'uint32_t' has not been declared
  src/net/third_party/quiche/src/quiche/quic/core/crypto/quic_hkdf.cc:39:3: error: 'output_' was not declared in this scope
  src/net/third_party/quiche/src/quiche/quic/core/crypto/quic_hkdf.h:58:15: error: 'uint8_t' was not declared in this scope
  src/net/third_party/quiche/src/quiche/quic/core/crypto/quic_hkdf.h:58:22: error: template argument 1 is invalid
  src/net/third_party/quiche/src/quiche/quic/core/crypto/quic_hkdf.h:58:22: error: template argument 2 is invalid
  src/net/third_party/quiche/src/quiche/quic/core/crypto/quic_hkdf.h:58:8: error: '<expression error>' in namespace 'std' does not name a type
  src/net/third_party/quiche/src/quiche/quic/core/deterministic_connection_id_generator.cc:17:7: error: 'kQuicMaxConnectionIdWithLengthPrefixLength' was not declared in this scope
  src/net/third_party/quiche/src/quiche/quic/core/deterministic_connection_id_generator.cc:30:51: error: 'const class quic::QuicConnectionId' has no member named 'length'; did you mean 'set_length'?
  src/net/third_party/quiche/src/quiche/quic/core/deterministic_connection_id_generator.cc:38:51: error: 'const class quic::QuicConnectionId' has no member named 'length'; did you mean 'set_length'?
  src/net/third_party/quiche/src/quiche/quic/core/deterministic_connection_id_generator.cc:56:16: error: 'const class quic::QuicConnectionId' has no member named 'length'; did you mean 'set_length'?
  src/net/third_party/quiche/src/quiche/quic/core/deterministic_connection_id_generator.cc:67:71: error: 'class quic::QuicConnectionId' has no member named 'length'; did you mean 'set_length'?
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:112:7: error: 'uint8_t' does not name a type
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:116:7: error: 'uint8_t' does not name a type
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:23:31: error: found ':' in nested-name-specifier, expected '::'
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:23:41: error: expected unqualified-id before '{' token
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:23:6: error: 'QuicConnectionIdIncluded' has not been declared
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:30:7: error: 'uint8_t' does not name a type
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:34:7: error: 'uint8_t' does not name a type
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:38:7: error: 'uint8_t' does not name a type
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:42:7: error: 'uint8_t' does not name a type
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:50:38: error: 'uint8_t' has not been declared
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:51:20: error: missing template argument list after 'absl::Span'; template placeholder not permitted in parameter
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:51:26: error: invalid template-id
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:51:37: error: expected '>' before 'const'
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:51:37: error: expected primary-expression before 'const'
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:51:43: error: ISO C++ forbids declaration of 'type name' with no type [-fpermissive]
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:51:50: error: template argument 1 is invalid
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:62:3: error: 'uint8_t' does not name a type
  src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h:68:19: error: 'uint8_t' has not been declared
  src/sandbox/linux/syscall_broker/broker_file_permission.cc:267:1: error: no declaration matches 'sandbox::syscall_broker::BrokerFilePermission::BrokerFilePermission(std::string, uint64_t)'
  src/sandbox/linux/syscall_broker/broker_file_permission.h:184:42: error: 'uint64_t' has not been declared
  src/sandbox/linux/syscall_broker/broker_file_permission.h:189:3: error: 'uint64_t' does not name a type
  src/third_party/angle/include/GLSLANG/ShaderVars.h:126:70: error: 'uint32_t' has not been declared
  src/third_party/angle/src/libANGLE/VaryingPacking.cpp:759:71: error: invalid conversion from 'GLuint*' {aka 'unsigned int*'} to 'int*' [-fpermissive]
  src/third_party/blink/common/bluetooth/web_bluetooth_device_id.cc:23:34: error: 'WebBluetoothDeviceIdKey' was not declared in this scope; did you mean 'WebBluetoothDeviceId'?
  src/third_party/blink/common/bluetooth/web_bluetooth_device_id.cc:24:32: error: 'device_id_' was not declared in this scope
  src/third_party/blink/common/bluetooth/web_bluetooth_device_id.cc:31:11: error: 'WebBluetoothDeviceIdKey' does not name a type; did you mean 'WebBluetoothDeviceId'?
  src/third_party/blink/common/bluetooth/web_bluetooth_device_id.cc:32:7: error: class 'blink::WebBluetoothDeviceId' does not have any field named 'device_id_'
  src/third_party/blink/common/bluetooth/web_bluetooth_device_id.cc:36:29: error: 'device_id_' was not declared in this scope
  src/third_party/blink/common/bluetooth/web_bluetooth_device_id.cc:43:7: error: 'WebBluetoothDeviceIdKey' does not name a type; did you mean 'WebBluetoothDeviceId'?
  src/third_party/blink/common/bluetooth/web_bluetooth_device_id.cc:50:3: error: 'WebBluetoothDeviceIdKey' was not declared in this scope; did you mean 'WebBluetoothDeviceId'?
  src/third_party/blink/common/bluetooth/web_bluetooth_device_id.cc:52:21: error: 'bytes' was not declared in this scope
  src/third_party/blink/common/bluetooth/web_bluetooth_device_id.cc:64:32: error: 'WebBluetoothDeviceIdKey' was not declared in this scope; did you mean 'WebBluetoothDeviceId'?
  src/third_party/blink/common/bluetooth/web_bluetooth_device_id.cc:77:16: error: 'const class blink::WebBluetoothDeviceId' has no member named 'DeviceId'
  src/third_party/blink/common/bluetooth/web_bluetooth_device_id.cc:77:40: error: 'const class blink::WebBluetoothDeviceId' has no member named 'DeviceId'
  src/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h:15:38: error: '<expression error>' in namespace 'std' does not name a type
  src/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h:15:44: error: 'uint8_t' was not declared in this scope
  src/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h:15:55: error: template argument 1 is invalid
  src/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h:28:39: error: 'WebBluetoothDeviceIdKey' does not name a type; did you mean 'WebBluetoothDeviceId'?
  src/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h:50:9: error: 'WebBluetoothDeviceIdKey' does not name a type; did you mean 'WebBluetoothDeviceId'?
  src/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h:67:3: error: 'WebBluetoothDeviceIdKey' does not name a type; did you mean 'WebBluetoothDeviceId'?
  src/third_party/blink/renderer/platform/graphics/paint/effect_paint_property_node.h:295:3: error: declaration of 'const cc::DocumentTransitionSharedElementId& blink::EffectPaintPropertyNode::DocumentTransitionSharedElementId() const' changes meaning of 'DocumentTransitionSharedElementId' [-Wchanges-meaning]
  src/third_party/dawn/src/dawn/native/stream/ByteVectorSink.h:26:43: error: 'uint8_t' was not declared in this scope
  src/third_party/dawn/src/dawn/native/stream/ByteVectorSink.h:26:50: error: template argument 1 is invalid
  src/third_party/dawn/src/dawn/native/stream/ByteVectorSink.h:26:50: error: template argument 2 is invalid
  src/third_party/dawn/src/dawn/native/stream/ByteVectorSink.h:28:23: error: 'uint8_t' was not declared in this scope
  src/third_party/dawn/src/dawn/native/stream/ByteVectorSink.h:28:30: error: template argument 1 is invalid
  src/third_party/dawn/src/dawn/native/stream/ByteVectorSink.h:28:30: error: template argument 2 is invalid
  src/third_party/dawn/src/tint/reader/spirv/namer.cc:106:19: error: 'next_unusued_derived_name_id_' was not declared in this scope
  src/third_party/dawn/src/tint/reader/spirv/namer.cc:130:6: error: no declaration matches 'bool tint::reader::spirv::Namer::Register(uint32_t, const std::string&)'
  src/third_party/dawn/src/tint/reader/spirv/namer.cc:147:5: error: 'name_to_id_' was not declared in this scope
  src/third_party/dawn/src/tint/reader/spirv/namer.cc:151:6: error: no declaration matches 'bool tint::reader::spirv::Namer::SuggestSanitizedName(uint32_t, const std::string&)'
  src/third_party/dawn/src/tint/reader/spirv/namer.cc:159:6: error: no declaration matches 'bool tint::reader::spirv::Namer::SuggestSanitizedMemberName(uint32_t, uint32_t, const std::string&)'
  src/third_party/dawn/src/tint/reader/spirv/namer.cc:174:6: error: no declaration matches 'void tint::reader::spirv::Namer::ResolveMemberNamesForStruct(uint32_t, uint32_t)'
  src/third_party/dawn/src/tint/reader/spirv/namer.cc:62:9: error: 'name_to_id_' was not declared in this scope
  src/third_party/dawn/src/tint/reader/spirv/namer.cc:91:13: error: no declaration matches 'std::string tint::reader::spirv::Namer::GetMemberName(uint32_t, uint32_t) const'
  src/third_party/dawn/src/tint/reader/spirv/namer.h:103:19: error: 'uint32_t' has not been declared
  src/third_party/dawn/src/tint/reader/spirv/namer.h:117:31: error: 'uint32_t' has not been declared
  src/third_party/dawn/src/tint/reader/spirv/namer.h:125:37: error: 'uint32_t' has not been declared
  src/third_party/dawn/src/tint/reader/spirv/namer.h:126:37: error: 'uint32_t' has not been declared
  src/third_party/dawn/src/tint/reader/spirv/namer.h:135:38: error: 'uint32_t' has not been declared
  src/third_party/dawn/src/tint/reader/spirv/namer.h:135:58: error: 'uint32_t' has not been declared
  src/third_party/dawn/src/tint/reader/spirv/namer.h:141:10: error: '<expression error>' in namespace 'std' does not name a type
  src/third_party/dawn/src/tint/reader/spirv/namer.h:141:24: error: 'uint32_t' was not declared in this scope
  src/third_party/dawn/src/tint/reader/spirv/namer.h:141:45: error: template argument 1 is invalid
  src/third_party/dawn/src/tint/reader/spirv/namer.h:141:45: error: template argument 3 is invalid
  src/third_party/dawn/src/tint/reader/spirv/namer.h:141:45: error: template argument 4 is invalid
  src/third_party/dawn/src/tint/reader/spirv/namer.h:141:45: error: template argument 5 is invalid
  src/third_party/dawn/src/tint/reader/spirv/namer.h:143:10: error: '<expression error>' in namespace 'std' does not name a type
  src/third_party/dawn/src/tint/reader/spirv/namer.h:143:37: error: 'uint32_t' was not declared in this scope
  src/third_party/dawn/src/tint/reader/spirv/namer.h:143:45: error: template argument 2 is invalid
  src/third_party/dawn/src/tint/reader/spirv/namer.h:143:45: error: template argument 5 is invalid
  src/third_party/dawn/src/tint/reader/spirv/namer.h:148:10: error: '<expression error>' in namespace 'std' does not name a type
  src/third_party/dawn/src/tint/reader/spirv/namer.h:148:24: error: 'uint32_t' was not declared in this scope
  src/third_party/dawn/src/tint/reader/spirv/namer.h:148:57: error: template argument 1 is invalid
  src/third_party/dawn/src/tint/reader/spirv/namer.h:148:57: error: template argument 3 is invalid
  src/third_party/dawn/src/tint/reader/spirv/namer.h:148:57: error: template argument 4 is invalid
  src/third_party/dawn/src/tint/reader/spirv/namer.h:148:57: error: template argument 5 is invalid
  src/third_party/dawn/src/tint/reader/spirv/namer.h:152:10: error: '<expression error>' in namespace 'std' does not name a type
  src/third_party/dawn/src/tint/reader/spirv/namer.h:152:37: error: 'uint32_t' was not declared in this scope
  src/third_party/dawn/src/tint/reader/spirv/namer.h:152:45: error: template argument 2 is invalid
  src/third_party/dawn/src/tint/reader/spirv/namer.h:152:45: error: template argument 5 is invalid
  src/third_party/dawn/src/tint/reader/spirv/namer.h:54:18: error: 'uint32_t' has not been declared
  src/third_party/dawn/src/tint/reader/spirv/namer.h:54:40: error: 'id_to_name_' was not declared in this scope
  src/third_party/dawn/src/tint/reader/spirv/namer.h:59:16: error: 'name_to_id_' was not declared in this scope
  src/third_party/dawn/src/tint/reader/spirv/namer.h:64:32: error: 'uint32_t' has not been declared
  src/third_party/dawn/src/tint/reader/spirv/namer.h:64:60: error: 'id_to_name_' was not declared in this scope
  src/third_party/dawn/src/tint/reader/spirv/namer.h:70:29: error: 'uint32_t' has not been declared
  src/third_party/dawn/src/tint/reader/spirv/namer.h:84:31: error: 'uint32_t' has not been declared
  src/third_party/dawn/src/tint/reader/spirv/namer.h:84:44: error: 'uint32_t' has not been declared
  src/third_party/openscreen/src/util/base64.cc:20:51: error: 'std::string openscreen::base64::Encode(absl::Span<const unsigned char>)' redeclared as different kind of entity
  src/third_party/openscreen/src/util/base64.cc:21:16: error: no match for call to '(std::string {aka std::__cxx11::basic_string<char>}) (absl::string_view)'
  src/third_party/openscreen/src/util/base64.cc:25:43: error: 'std::string openscreen::base64::Encode(absl::string_view)' redeclared as different kind of entity
  src/third_party/openscreen/src/util/base64.h:19:20: error: invalid template-id
  src/third_party/openscreen/src/util/base64.h:19:37: error: ISO C++ forbids declaration of 'type name' with no type [-fpermissive]
  src/third_party/openscreen/src/util/base64.h:19:44: error: template argument 1 is invalid
  src/third_party/openscreen/src/util/base64.h:19:46: error: missing template arguments before 'input'
  src/third_party/openscreen/src/util/base64.h:22:43: error: 'std::string openscreen::base64::Encode(absl::string_view)' redeclared as different kind of entity
  src/third_party/openscreen/src/util/base64.h:27:50: error: 'uint8_t' was not declared in this scope
  src/third_party/openscreen/src/util/base64.h:27:57: error: template argument 1 is invalid
  src/third_party/openscreen/src/util/base64.h:27:57: error: template argument 2 is invalid
  src/third_party/perfetto/include/perfetto/ext/base/uuid.h:34:14: error: 'uint8_t' was not declared in this scope
  src/third_party/perfetto/include/perfetto/ext/base/uuid.h:34:25: error: template argument 1 is invalid
  src/third_party/perfetto/include/perfetto/ext/base/uuid.h:34:8: error: '<expression error>' in namespace 'std' does not name a type
  src/third_party/perfetto/include/perfetto/ext/base/uuid.h:35:14: error: '<expression error>' in namespace 'std' does not name a type
  src/third_party/perfetto/include/perfetto/ext/base/uuid.h:35:20: error: 'uint8_t' was not declared in this scope
  src/third_party/perfetto/include/perfetto/ext/base/uuid.h:35:31: error: template argument 1 is invalid
  src/third_party/perfetto/include/perfetto/ext/base/uuid.h:37:53: error: 'data_' was not declared in this scope
  src/third_party/perfetto/include/perfetto/ext/base/uuid.h:37:68: error: 'const class perfetto::base::Uuid' has no member named 'data_'
  src/third_party/perfetto/include/perfetto/ext/base/uuid.h:43:21: error: 'data_' was not declared in this scope
  src/third_party/perfetto/include/perfetto/ext/base/uuid.h:49:21: error: 'data_' was not declared in this scope
  src/third_party/perfetto/include/perfetto/ext/base/uuid.h:57:38: error: 'data_' was not declared in this scope
  src/third_party/perfetto/include/perfetto/ext/base/uuid.h:58:38: error: 'data_' was not declared in this scope
  src/third_party/perfetto/include/perfetto/ext/base/uuid.h:64:14: error: 'uint8_t' was not declared in this scope
  src/third_party/perfetto/include/perfetto/ext/base/uuid.h:64:25: error: template argument 1 is invalid
  src/third_party/perfetto/include/perfetto/ext/base/uuid.h:64:8: error: '<expression error>' in namespace 'std' does not name a type
  src/third_party/perfetto/src/base/uuid.cc:35:22: error: 'class perfetto::base::Uuid' has no member named 'data'
  src/third_party/perfetto/src/base/uuid.cc:51:30: error: 'data_' was not declared in this scope
  src/third_party/perfetto/src/base/uuid.cc:52:10: error: 'data_' was not declared in this scope
  src/third_party/perfetto/src/base/uuid.cc:60:52: error: 'data_' was not declared in this scope
  src/third_party/perfetto/src/base/uuid.cc:64:17: error: 'data_' was not declared in this scope
  src/third_party/ruy/src/ruy/profiler/instrumentation.cc:34:12: error: 'string' is not a member of 'std'
  src/third_party/ruy/src/ruy/profiler/instrumentation.cc:34:36: error: 'string' is not a member of 'std'
  src/third_party/ruy/src/ruy/profiler/instrumentation.cc:48:6: error: 'string' in namespace 'std' does not name a type
  src/third_party/ruy/src/ruy/profiler/instrumentation.h:147:7: error: there are no arguments to 'abort' that depend on a template parameter, so a declaration of 'abort' must be available [-fpermissive]
  src/third_party/ruy/src/ruy/profiler/instrumentation.h:55:8: error: 'string' in namespace 'std' does not name a type
  src/third_party/swiftshader/src/System/LRUCache.hpp:116:24: error: 'uint64_t' does not name a type
  src/third_party/swiftshader/src/System/LRUCache.hpp:119:24: error: 'uint64_t' does not name a type
  src/third_party/swiftshader/src/System/LRUCache.hpp:344:1: error: 'uint64_t' does not name a type
  src/third_party/swiftshader/src/System/LRUCache.hpp:350:1: error: 'uint64_t' does not name a type
  src/third_party/tflite/src/tensorflow/lite/kernels/internal/spectrogram.cc:46:22: error: 'uint32_t' was not declared in this scope
  src/third_party/tflite/src/tensorflow/lite/kernels/internal/spectrogram.cc:61:24: error: 'uint32_t' was not declared in this scope
  src/third_party/tflite/src/tensorflow/lite/kernels/internal/spectrogram.cc:69:8: error: 'uint32_t' does not name a type
  src/third_party/tflite/src/tensorflow/lite/kernels/internal/spectrogram.cc:92:17: error: 'NextPowerOfTwo' was not declared in this scope
  src/third_party/vulkan_memory_allocator/include/vk_mem_alloc.h:2633:9: error: 'snprintf' was not declared in this scope
  src/third_party/vulkan_memory_allocator/include/vk_mem_alloc.h:2637:9: error: 'snprintf' was not declared in this scope
  src/third_party/vulkan_memory_allocator/include/vk_mem_alloc.h:2641:9: error: 'snprintf' was not declared in this scope
  src/third_party/webrtc/rtc_base/third_party/base64/base64.cc:213:6: error: no declaration matches 'bool rtc::Base64::DecodeFromArray(const char*, size_t, DecodeFlags, std::vector<unsigned char>*, size_t*)'
  src/third_party/webrtc/rtc_base/third_party/base64/base64.h:78:43: error: 'uint8_t' was not declared in this scope
  src/third_party/webrtc/rtc_base/third_party/base64/base64.h:78:50: error: template argument 1 is invalid
  src/third_party/webrtc/rtc_base/third_party/base64/base64.h:78:50: error: template argument 2 is invalid
  src/third_party/webrtc/rtc_base/third_party/base64/base64.h:78:51: error: expected identifier before '*' token
  src/ui/base/prediction/kalman_filter.cc:10:11: error: 'uint32_t' does not name a type
  src/ui/base/prediction/kalman_filter.cc:111:13: error: 'iteration_count_' was not declared in this scope
  src/ui/base/prediction/kalman_filter.cc:45:7: error: class 'ui::KalmanFilter' does not have any field named 'iteration_count_'
  src/ui/base/prediction/kalman_filter.cc:54:10: error: 'iteration_count_' was not declared in this scope
  src/ui/base/prediction/kalman_filter.cc:54:30: error: 'kStableIterNum' was not declared in this scope
  src/ui/base/prediction/kalman_filter.cc:58:7: error: 'iteration_count_' was not declared in this scope
  src/ui/base/prediction/kalman_filter.cc:95:3: error: 'iteration_count_' was not declared in this scope
  src/ui/base/prediction/kalman_filter.h:82:3: error: 'uint32_t' does not name a type
  src/ui/events/gesture_event_details.cc:30:42: error: 'ui::ScrollGranularity' has not been declared
  src/ui/events/gesture_event_details.cc:40:26: error: 'struct ui::GestureEventDetails::Details::<unnamed>' has no member named 'delta_hint_units'
  src/ui/events/gesture_event_details.cc:46:27: error: 'struct ui::GestureEventDetails::Details::<unnamed>' has no member named 'delta_units'
  src/ui/events/gesture_event_details.h:106:7: error: 'ScrollGranularity' in namespace 'ui' does not name a type
  src/ui/events/gesture_event_details.h:195:11: error: 'ScrollGranularity' in namespace 'ui' does not name a type
  src/ui/events/gesture_event_details.h:201:11: error: 'ScrollGranularity' in namespace 'ui' does not name a type
  src/ui/events/gesture_event_details.h:29:23: error: 'ui::ScrollGranularity' has not been declared
  src/ui/events/gesture_event_details.h:30:31: error: 'ui::ScrollGranularity' has not been declared
  src/ui/events/gesture_event_details.h:91:7: error: 'ScrollGranularity' in namespace 'ui' does not name a type
  src/ui/events/types/scroll_types.h:10:12: error: 'ScrollGranularity' has not been declared
  src/ui/events/types/scroll_types.h:10:30: error: found ':' in nested-name-specifier, expected '::'
  src/ui/events/types/scroll_types.h:10:40: error: expected unqualified-id before '{' token
  src/ui/gfx/geometry/linear_gradient.cc:32:6: error: no declaration matches 'void gfx::LinearGradient::AddStep(float, uint8_t)'
  src/ui/gfx/geometry/linear_gradient.cc:83:73: error: 'const std::array<gfx::LinearGradient::Step, 8>::value_type' {aka 'const struct gfx::LinearGradient::Step'} has no member named 'alpha'
  src/ui/gfx/geometry/linear_gradient.h:33:5: error: 'uint8_t' does not name a type
  src/ui/gfx/geometry/linear_gradient.h:48:32: error: 'uint8_t' has not been declared
  src/ui/gfx/geometry/linear_gradient.h:77:46: error: 'const struct gfx::LinearGradient::Step' has no member named 'alpha'
  src/ui/gfx/geometry/linear_gradient.h:77:59: error: 'const struct gfx::LinearGradient::Step' has no member named 'alpha'
  lib32-recipe-sysroot/usr/include/c++/13.0.1/bits/hashtable_policy.h:1304:23: error: static assertion failed: hash function must be invocable with an argument of key type
  lib32-recipe-sysroot/usr/include/c++/13.0.1/bits/hashtable_policy.h:1306:25: error: no match for call to '(const sw::LRUCache<sw::Blitter::State, rr::RoutineT<void(const sw::Blitter::BlitData*)>, std::hash<sw::Blitter::State> >::KeyedComparator) (const sw::LRUCache<sw::Blitter::State, rr::RoutineT<void(const sw::Blitter::BlitData*)>, std::hash<sw::Blitter::State> >::Keyed* const&)'
  lib32-recipe-sysroot/usr/include/c++/13.0.1/bits/hashtable_policy.h:1306:25: error: no match for call to '(const sw::LRUCache<sw::Blitter::State, rr::RoutineT<void(const sw::Blitter::CubeBorderData*)>, std::hash<sw::Blitter::State> >::KeyedComparator) (const sw::LRUCache<sw::Blitter::State, rr::RoutineT<void(const sw::Blitter::CubeBorderData*)>, std::hash<sw::Blitter::State> >::Keyed* const&)'
  lib32-recipe-sysroot/usr/include/c++/13.0.1/bits/hashtable_policy.h:1721:23: error: static assertion failed: key equality predicate must be invocable with two arguments of key type
  lib32-recipe-sysroot/usr/include/c++/13.0.1/bits/hashtable_policy.h:1724:23: error: no match for call to '(const sw::LRUCache<sw::Blitter::State, rr::RoutineT<void(const sw::Blitter::BlitData*)>, std::hash<sw::Blitter::State> >::KeyedComparator) (const sw::LRUCache<sw::Blitter::State, rr::RoutineT<void(const sw::Blitter::BlitData*)>, std::hash<sw::Blitter::State> >::Keyed* const&, const sw::LRUCache<sw::Blitter::State, rr::RoutineT<void(const sw::Blitter::BlitData*)>, std::hash<sw::Blitter::State> >::Keyed* const&)'
  lib32-webruntime/108.0.5359.217-pro.5-r64.0/lib32-recipe-sysroot/usr/include/c++/13.0.1/bits/hashtable_policy.h:1724:23: error: no match for call to '(const sw::LRUCache<sw::Blitter::State, rr::RoutineT<void(const sw::Blitter::CubeBorderData*)>, std::hash<sw::Blitter::State> >::KeyedComparator) (const sw::LRUCache<sw::Blitter::State, rr::RoutineT<void(const sw::Blitter::CubeBorderData*)>, std::hash<sw::Blitter::State> >::Keyed* const&, const sw::LRUCache<sw::Blitter::State, rr::RoutineT<void(const sw::Blitter::CubeBorderData*)>, std::hash<sw::Blitter::State> >::Keyed* const&)'

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRP-310] Apply .patch files for hot-fix to corresponding repositories
[WRP-13301] Create GPVB with Yocto 4.3 Nanbield

Change-Id: I9ad64589cc33624a9400196e47b4ef65aebfbec8
Reviewed-on: http://gpro.lge.com/c/webosose/chromium108/+/351386
Reviewed-by: Commit Msg Checker <commit_msg@lge.com>
Reviewed-by: Jose Dapena Paz <j.dapena@lgepartner.com>
Reviewed-by: Donghyun Kim <donghyun11.kim@lge.com>
Tested-by: Donghyun Kim <donghyun11.kim@lge.com>
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/chromium108/+/373249 [op][upsable][build] Fix build with gcc-13 (master branch) AND http://gpro.lge.com/c/webosose/chromium108/+/351386 [op][upsable][build] Fix build with gcc-13 (@pro branch)]

 src/base/debug/profiler.h                                       | 1 +
 src/cc/trees/target_property.cc                                 | 2 ++
 src/components/autofill/core/browser/autofill_ablation_study.h  | 1 +
 .../feature_engagement/internal/event_storage_validator.h       | 1 +
 src/components/metrics/psi_memory_parser.h                      | 1 +
 .../core/browser/generation/password_generator.h                | 1 +
 src/components/viz/common/shared_element_resource_id.h          | 1 +
 src/device/base/synchronization/one_writer_seqlock.h            | 1 +
 src/device/bluetooth/public/cpp/bluetooth_uuid.h                | 1 +
 src/extensions/common/constants.h                               | 2 ++
 src/extensions/renderer/bindings/api_invocation_errors.h        | 1 +
 src/gpu/config/gpu_feature_info.h                               | 1 +
 src/net/base/parse_number.h                                     | 2 ++
 src/net/cookies/cookie_inclusion_status.h                       | 1 +
 .../third_party/quiche/src/quiche/quic/core/crypto/quic_hkdf.h  | 1 +
 .../quiche/src/quiche/quic/core/quic_connection_id.h            | 1 +
 src/sandbox/linux/syscall_broker/broker_file_permission.h       | 1 +
 src/third_party/angle/include/GLSLANG/ShaderVars.h              | 1 +
 .../blink/public/common/bluetooth/web_bluetooth_device_id.h     | 1 +
 .../platform/graphics/paint/effect_paint_property_node.h        | 2 +-
 src/third_party/dawn/src/dawn/native/stream/ByteVectorSink.h    | 1 +
 src/third_party/dawn/src/tint/reader/spirv/namer.h              | 1 +
 .../openscreen/src/discovery/dnssd/public/dns_sd_txt_record.h   | 1 +
 src/third_party/openscreen/src/util/base64.h                    | 1 +
 src/third_party/perfetto/include/perfetto/ext/base/uuid.h       | 1 +
 src/third_party/ruy/src/ruy/profiler/instrumentation.h          | 1 +
 src/third_party/swiftshader/src/System/LRUCache.hpp             | 1 +
 .../tflite/src/tensorflow/lite/kernels/internal/spectrogram.cc  | 1 +
 src/third_party/vulkan_memory_allocator/include/vk_mem_alloc.h  | 1 +
 src/third_party/webrtc/rtc_base/third_party/base64/base64.h     | 1 +
 src/ui/base/prediction/kalman_filter.h                          | 2 ++
 src/ui/events/gesture_event_details.h                           | 1 +
 src/ui/events/types/scroll_types.h                              | 2 ++
 src/ui/gfx/geometry/linear_gradient.h                           | 1 +
 34 files changed, 39 insertions(+), 1 deletion(-)

diff --git a/src/base/debug/profiler.h b/src/base/debug/profiler.h
index 5f0deadd5d..7d3eeaf35f 100644
--- a/src/base/debug/profiler.h
+++ b/src/base/debug/profiler.h
@@ -7,6 +7,7 @@
 
 #include <stddef.h>
 
+#include <cstdint>
 #include <string>
 
 #include "base/base_export.h"
diff --git a/src/cc/trees/target_property.cc b/src/cc/trees/target_property.cc
index f6e873a01a..5f952c5ed0 100644
--- a/src/cc/trees/target_property.cc
+++ b/src/cc/trees/target_property.cc
@@ -6,6 +6,8 @@
 
 #include "ui/gfx/animation/keyframe/target_property.h"
 
+#include <cstdint>
+
 namespace cc {
 
 static_assert(TargetProperty::LAST_TARGET_PROPERTY <
diff --git a/src/components/autofill/core/browser/autofill_ablation_study.h b/src/components/autofill/core/browser/autofill_ablation_study.h
index ada8121982..35cce67d5c 100644
--- a/src/components/autofill/core/browser/autofill_ablation_study.h
+++ b/src/components/autofill/core/browser/autofill_ablation_study.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_AUTOFILL_CORE_BROWSER_AUTOFILL_ABLATION_STUDY_H_
 #define COMPONENTS_AUTOFILL_CORE_BROWSER_AUTOFILL_ABLATION_STUDY_H_
 
+#include <cstdint>
 #include <string>
 
 class GURL;
diff --git a/src/components/feature_engagement/internal/event_storage_validator.h b/src/components/feature_engagement/internal/event_storage_validator.h
index 647a359ee2..ed85a276be 100644
--- a/src/components/feature_engagement/internal/event_storage_validator.h
+++ b/src/components/feature_engagement/internal/event_storage_validator.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_FEATURE_ENGAGEMENT_INTERNAL_EVENT_STORAGE_VALIDATOR_H_
 #define COMPONENTS_FEATURE_ENGAGEMENT_INTERNAL_EVENT_STORAGE_VALIDATOR_H_
 
+#include <cstdint>
 #include <string>
 
 namespace feature_engagement {
diff --git a/src/components/metrics/psi_memory_parser.h b/src/components/metrics/psi_memory_parser.h
index b8b22f4ab7..bf02be5074 100644
--- a/src/components/metrics/psi_memory_parser.h
+++ b/src/components/metrics/psi_memory_parser.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_METRICS_PSI_MEMORY_PARSER_H_
 #define COMPONENTS_METRICS_PSI_MEMORY_PARSER_H_
 
+#include <cstdint>
 #include <string>
 
 #include "base/gtest_prod_util.h"
diff --git a/src/components/password_manager/core/browser/generation/password_generator.h b/src/components/password_manager/core/browser/generation/password_generator.h
index b0d621ce35..aa674d4779 100644
--- a/src/components/password_manager/core/browser/generation/password_generator.h
+++ b/src/components/password_manager/core/browser/generation/password_generator.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_PASSWORD_MANAGER_CORE_BROWSER_GENERATION_PASSWORD_GENERATOR_H_
 #define COMPONENTS_PASSWORD_MANAGER_CORE_BROWSER_GENERATION_PASSWORD_GENERATOR_H_
 
+#include <cstdint>
 #include <string>
 
 
diff --git a/src/components/viz/common/shared_element_resource_id.h b/src/components/viz/common/shared_element_resource_id.h
index b39a2d3dc1..ca3d8aedae 100644
--- a/src/components/viz/common/shared_element_resource_id.h
+++ b/src/components/viz/common/shared_element_resource_id.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_VIZ_COMMON_SHARED_ELEMENT_RESOURCE_ID_H_
 #define COMPONENTS_VIZ_COMMON_SHARED_ELEMENT_RESOURCE_ID_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
diff --git a/src/device/base/synchronization/one_writer_seqlock.h b/src/device/base/synchronization/one_writer_seqlock.h
index 528b6683ca..e37a16b81b 100644
--- a/src/device/base/synchronization/one_writer_seqlock.h
+++ b/src/device/base/synchronization/one_writer_seqlock.h
@@ -6,6 +6,7 @@
 #define DEVICE_BASE_SYNCHRONIZATION_ONE_WRITER_SEQLOCK_H_
 
 #include <atomic>
+#include <cstdint>
 #include <type_traits>
 
 #include "base/atomicops.h"
diff --git a/src/device/bluetooth/public/cpp/bluetooth_uuid.h b/src/device/bluetooth/public/cpp/bluetooth_uuid.h
index 816d3405eb..15c4004c01 100644
--- a/src/device/bluetooth/public/cpp/bluetooth_uuid.h
+++ b/src/device/bluetooth/public/cpp/bluetooth_uuid.h
@@ -5,6 +5,7 @@
 #ifndef DEVICE_BLUETOOTH_PUBLIC_CPP_BLUETOOTH_UUID_H_
 #define DEVICE_BLUETOOTH_PUBLIC_CPP_BLUETOOTH_UUID_H_
 
+#include <cstdint>
 #include <ostream>
 #include <string>
 #include <vector>
diff --git a/src/extensions/common/constants.h b/src/extensions/common/constants.h
index bf6bac2b47..51c5345091 100644
--- a/src/extensions/common/constants.h
+++ b/src/extensions/common/constants.h
@@ -10,6 +10,8 @@
 #include "build/chromeos_buildflags.h"
 #include "extensions/common/extensions_export.h"
 
+#include <cstdint>
+
 namespace extensions {
 
 // Scheme we serve extension content from.
diff --git a/src/extensions/renderer/bindings/api_invocation_errors.h b/src/extensions/renderer/bindings/api_invocation_errors.h
index e21700ecfc..20492657d3 100644
--- a/src/extensions/renderer/bindings/api_invocation_errors.h
+++ b/src/extensions/renderer/bindings/api_invocation_errors.h
@@ -6,6 +6,7 @@
 #define EXTENSIONS_RENDERER_BINDINGS_API_INVOCATION_ERRORS_H_
 
 #include <set>
+#include <cstdint>
 #include <string>
 
 #include "base/strings/string_piece.h"
diff --git a/src/gpu/config/gpu_feature_info.h b/src/gpu/config/gpu_feature_info.h
index 1a7028e471..b9e951c0a4 100644
--- a/src/gpu/config/gpu_feature_info.h
+++ b/src/gpu/config/gpu_feature_info.h
@@ -5,6 +5,7 @@
 #ifndef GPU_CONFIG_GPU_FEATURE_INFO_H_
 #define GPU_CONFIG_GPU_FEATURE_INFO_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
diff --git a/src/net/base/parse_number.h b/src/net/base/parse_number.h
index d6adf388e6..560e692aab 100644
--- a/src/net/base/parse_number.h
+++ b/src/net/base/parse_number.h
@@ -8,6 +8,8 @@
 #include "base/strings/string_piece.h"
 #include "net/base/net_export.h"
 
+#include <cstdint>
+
 // This file contains utility functions for parsing numbers, in the context of
 // network protocols.
 //
diff --git a/src/net/cookies/cookie_inclusion_status.h b/src/net/cookies/cookie_inclusion_status.h
index 5f289ddcd9..44f2f9eb1f 100644
--- a/src/net/cookies/cookie_inclusion_status.h
+++ b/src/net/cookies/cookie_inclusion_status.h
@@ -6,6 +6,7 @@
 #define NET_COOKIES_COOKIE_INCLUSION_STATUS_H_
 
 #include <bitset>
+#include <cstdint>
 #include <ostream>
 #include <string>
 #include <vector>
diff --git a/src/net/third_party/quiche/src/quiche/quic/core/crypto/quic_hkdf.h b/src/net/third_party/quiche/src/quiche/quic/core/crypto/quic_hkdf.h
index 6a300ed0c0..3e30f1cb02 100644
--- a/src/net/third_party/quiche/src/quiche/quic/core/crypto/quic_hkdf.h
+++ b/src/net/third_party/quiche/src/quiche/quic/core/crypto/quic_hkdf.h
@@ -5,6 +5,7 @@
 #ifndef QUICHE_QUIC_CORE_CRYPTO_QUIC_HKDF_H_
 #define QUICHE_QUIC_CORE_CRYPTO_QUIC_HKDF_H_
 
+#include <cstdint>
 #include <vector>
 
 #include "absl/strings/string_view.h"
diff --git a/src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h b/src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h
index 52f1501728..db4d310849 100644
--- a/src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h
+++ b/src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h
@@ -5,6 +5,7 @@
 #ifndef QUICHE_QUIC_CORE_QUIC_CONNECTION_ID_H_
 #define QUICHE_QUIC_CORE_QUIC_CONNECTION_ID_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
diff --git a/src/sandbox/linux/syscall_broker/broker_file_permission.h b/src/sandbox/linux/syscall_broker/broker_file_permission.h
index b167ffb099..2347d54616 100644
--- a/src/sandbox/linux/syscall_broker/broker_file_permission.h
+++ b/src/sandbox/linux/syscall_broker/broker_file_permission.h
@@ -6,6 +6,7 @@
 #define SANDBOX_LINUX_SYSCALL_BROKER_BROKER_FILE_PERMISSION_H_
 
 #include <bitset>
+#include <cstdint>
 #include <string>
 
 #include "sandbox/sandbox_export.h"
diff --git a/src/third_party/angle/include/GLSLANG/ShaderVars.h b/src/third_party/angle/include/GLSLANG/ShaderVars.h
index 4b76d33af2..4db1902d1b 100644
--- a/src/third_party/angle/include/GLSLANG/ShaderVars.h
+++ b/src/third_party/angle/include/GLSLANG/ShaderVars.h
@@ -12,6 +12,7 @@
 
 #include <algorithm>
 #include <array>
+#include <cstdint>
 #include <string>
 #include <vector>
 
diff --git a/src/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h b/src/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h
index 7a74887af9..058870eef3 100644
--- a/src/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h
+++ b/src/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h
@@ -6,6 +6,7 @@
 #define THIRD_PARTY_BLINK_PUBLIC_COMMON_BLUETOOTH_WEB_BLUETOOTH_DEVICE_ID_H_
 
 #include <array>
+#include <cstdint>
 #include <string>
 
 #include "third_party/blink/public/common/common_export.h"
diff --git a/src/third_party/blink/renderer/platform/graphics/paint/effect_paint_property_node.h b/src/third_party/blink/renderer/platform/graphics/paint/effect_paint_property_node.h
index e40a762079..be7c2a458e 100644
--- a/src/third_party/blink/renderer/platform/graphics/paint/effect_paint_property_node.h
+++ b/src/third_party/blink/renderer/platform/graphics/paint/effect_paint_property_node.h
@@ -122,7 +122,7 @@ class PLATFORM_EXPORT EffectPaintPropertyNode
 
     // An identifier for a document transition shared element. `id.valid()`
     // returns true if this has been set, and false otherwise.
-    DocumentTransitionSharedElementId document_transition_shared_element_id;
+    cc::DocumentTransitionSharedElementId document_transition_shared_element_id;
 
     // An identifier to tag shared element resources generated and cached in the
     // Viz process. This generated resource can be used as content for other
diff --git a/src/third_party/dawn/src/dawn/native/stream/ByteVectorSink.h b/src/third_party/dawn/src/dawn/native/stream/ByteVectorSink.h
index 3b6016c082..5537820b1a 100644
--- a/src/third_party/dawn/src/dawn/native/stream/ByteVectorSink.h
+++ b/src/third_party/dawn/src/dawn/native/stream/ByteVectorSink.h
@@ -15,6 +15,7 @@
 #ifndef SRC_DAWN_NATIVE_STREAM_BYTEVECTORSINK_H_
 #define SRC_DAWN_NATIVE_STREAM_BYTEVECTORSINK_H_
 
+#include <cstdint>
 #include <ostream>
 #include <vector>
 
diff --git a/src/third_party/dawn/src/tint/reader/spirv/namer.h b/src/third_party/dawn/src/tint/reader/spirv/namer.h
index 7a20e8738b..655f69ce36 100644
--- a/src/third_party/dawn/src/tint/reader/spirv/namer.h
+++ b/src/third_party/dawn/src/tint/reader/spirv/namer.h
@@ -15,6 +15,7 @@
 #ifndef SRC_TINT_READER_SPIRV_NAMER_H_
 #define SRC_TINT_READER_SPIRV_NAMER_H_
 
+#include <cstdint>
 #include <string>
 #include <unordered_map>
 #include <vector>
diff --git a/src/third_party/openscreen/src/discovery/dnssd/public/dns_sd_txt_record.h b/src/third_party/openscreen/src/discovery/dnssd/public/dns_sd_txt_record.h
index edf2c782f0..c23a92747b 100644
--- a/src/third_party/openscreen/src/discovery/dnssd/public/dns_sd_txt_record.h
+++ b/src/third_party/openscreen/src/discovery/dnssd/public/dns_sd_txt_record.h
@@ -5,6 +5,7 @@
 #ifndef DISCOVERY_DNSSD_PUBLIC_DNS_SD_TXT_RECORD_H_
 #define DISCOVERY_DNSSD_PUBLIC_DNS_SD_TXT_RECORD_H_
 
+#include <cstdint>
 #include <functional>
 #include <map>
 #include <set>
diff --git a/src/third_party/openscreen/src/util/base64.h b/src/third_party/openscreen/src/util/base64.h
index a7af9ecaa2..ad009a44ae 100644
--- a/src/third_party/openscreen/src/util/base64.h
+++ b/src/third_party/openscreen/src/util/base64.h
@@ -5,6 +5,7 @@
 #ifndef UTIL_BASE64_H_
 #define UTIL_BASE64_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
diff --git a/src/third_party/perfetto/include/perfetto/ext/base/uuid.h b/src/third_party/perfetto/include/perfetto/ext/base/uuid.h
index 1b4c53815b..472042fabc 100644
--- a/src/third_party/perfetto/include/perfetto/ext/base/uuid.h
+++ b/src/third_party/perfetto/include/perfetto/ext/base/uuid.h
@@ -18,6 +18,7 @@
 #define INCLUDE_PERFETTO_EXT_BASE_UUID_H_
 
 #include <array>
+#include <cstdint>
 #include <string>
 
 #include "perfetto/ext/base/optional.h"
diff --git a/src/third_party/ruy/src/ruy/profiler/instrumentation.h b/src/third_party/ruy/src/ruy/profiler/instrumentation.h
index c4df1e68fb..2b15ae369c 100644
--- a/src/third_party/ruy/src/ruy/profiler/instrumentation.h
+++ b/src/third_party/ruy/src/ruy/profiler/instrumentation.h
@@ -19,6 +19,7 @@ limitations under the License.
 #ifdef RUY_PROFILER
 #include <cstdio>
 #include <mutex>
+#include <string>
 #include <vector>
 #endif
 
diff --git a/src/third_party/swiftshader/src/System/LRUCache.hpp b/src/third_party/swiftshader/src/System/LRUCache.hpp
index 1c24490644..a04cdfafbc 100644
--- a/src/third_party/swiftshader/src/System/LRUCache.hpp
+++ b/src/third_party/swiftshader/src/System/LRUCache.hpp
@@ -18,6 +18,7 @@
 #include "System/Debug.hpp"
 
 #include <cstddef>
+#include <cstdint>
 #include <functional>
 #include <unordered_set>
 #include <vector>
diff --git a/src/third_party/tflite/src/tensorflow/lite/kernels/internal/spectrogram.cc b/src/third_party/tflite/src/tensorflow/lite/kernels/internal/spectrogram.cc
index a832962a38..3870071664 100644
--- a/src/third_party/tflite/src/tensorflow/lite/kernels/internal/spectrogram.cc
+++ b/src/third_party/tflite/src/tensorflow/lite/kernels/internal/spectrogram.cc
@@ -16,6 +16,7 @@ limitations under the License.
 #include "tensorflow/lite/kernels/internal/spectrogram.h"
 
 #include <assert.h>
+#include <cstdint>
 #include <math.h>
 
 #include "third_party/fft2d/fft.h"
diff --git a/src/third_party/vulkan_memory_allocator/include/vk_mem_alloc.h b/src/third_party/vulkan_memory_allocator/include/vk_mem_alloc.h
index b7c586b80e..e74b70ff1d 100644
--- a/src/third_party/vulkan_memory_allocator/include/vk_mem_alloc.h
+++ b/src/third_party/vulkan_memory_allocator/include/vk_mem_alloc.h
@@ -2389,6 +2389,7 @@ VMA_CALL_PRE void VMA_CALL_POST vmaFreeVirtualBlockStatsString(VmaVirtualBlock V
 #undef VMA_IMPLEMENTATION
 
 #include <cstdint>
+#include <cstdio>
 #include <cstdlib>
 #include <cstring>
 #include <utility>
diff --git a/src/third_party/webrtc/rtc_base/third_party/base64/base64.h b/src/third_party/webrtc/rtc_base/third_party/base64/base64.h
index 4190a79dc0..5a9285f007 100644
--- a/src/third_party/webrtc/rtc_base/third_party/base64/base64.h
+++ b/src/third_party/webrtc/rtc_base/third_party/base64/base64.h
@@ -12,6 +12,7 @@
 #ifndef RTC_BASE_THIRD_PARTY_BASE64_BASE64_H_
 #define RTC_BASE_THIRD_PARTY_BASE64_BASE64_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
diff --git a/src/ui/base/prediction/kalman_filter.h b/src/ui/base/prediction/kalman_filter.h
index bee3f946e9..f62e493b71 100644
--- a/src/ui/base/prediction/kalman_filter.h
+++ b/src/ui/base/prediction/kalman_filter.h
@@ -8,6 +8,8 @@
 #include "base/component_export.h"
 #include "ui/gfx/geometry/matrix3_f.h"
 
+#include <cstdint>
+
 namespace ui {
 
 // This Kalman filter is used to predict state in one axles.
diff --git a/src/ui/events/gesture_event_details.h b/src/ui/events/gesture_event_details.h
index 41ff9906d8..1864e9e21c 100644
--- a/src/ui/events/gesture_event_details.h
+++ b/src/ui/events/gesture_event_details.h
@@ -5,6 +5,7 @@
 #ifndef UI_EVENTS_GESTURE_EVENT_DETAILS_H_
 #define UI_EVENTS_GESTURE_EVENT_DETAILS_H_
 
+#include <cstdint>
 #include <string.h>
 
 #include "base/check_op.h"
diff --git a/src/ui/events/types/scroll_types.h b/src/ui/events/types/scroll_types.h
index be6b2779f4..dbe2e5cf74 100644
--- a/src/ui/events/types/scroll_types.h
+++ b/src/ui/events/types/scroll_types.h
@@ -5,6 +5,8 @@
 #ifndef UI_EVENTS_TYPES_SCROLL_TYPES_H_
 #define UI_EVENTS_TYPES_SCROLL_TYPES_H_
 
+#include <cstdint>
+
 namespace ui {
 
 enum class ScrollGranularity : uint8_t {
diff --git a/src/ui/gfx/geometry/linear_gradient.h b/src/ui/gfx/geometry/linear_gradient.h
index 65cd4cb07c..ab0ce5808c 100644
--- a/src/ui/gfx/geometry/linear_gradient.h
+++ b/src/ui/gfx/geometry/linear_gradient.h
@@ -6,6 +6,7 @@
 #define UI_GFX_LINEAR_GRADIENT_H_
 
 #include <array>
+#include <cstdint>
 #include <string>
 
 #include "ui/gfx/geometry/geometry_skia_export.h"
